File: scala-stuff/src/main/scala/CandlestickUsage.scala
import scala.util.Random
import Candlestick._

object CandlestickUsage extends App {
	val stocks = List("PETR4", "VALE5", "CSNA3")
	val random = new Random(System.nanoTime)
	val trades = {
		for(symbol <- stocks; 
			howMuch <- 1 to 5; 
			value <- 1 to 5)
			yield Trade(stock = symbol, 
						  quantity = math.abs(random.nextInt % 1000), 
						  price = math.abs(random.nextDouble * 100.0))
	}
	println("List of stocks:")
	trades.foreach(println)
	println("List of Candlesticks:")
	candlesticks(trades).foreach(println)	
}

File: scala-stuff/src/main/java/CandlestickUsage.java
import java.util.ArrayList;
import java.util.List;
import java.util.Random;


public class CandlestickUsage {


	public static void main(String[] args) {
		String [] acoes = new String[]{"PETR4", "VALUE5", "CSNA3"};
		Random random = new Random(System.nanoTime());
		List<Trade> trades = new ArrayList<Trade>();
		for(String acao : acoes){
			for(int i = 0; i < 5; i++){
				for(int j = 0; j < 5; j++){
					trades.add(Trade.createTrade(acao, 
							Math.abs(random.nextDouble() * 100.0), 
							Math.abs(random.nextInt() % 100)));
				}
			}
		}
		System.out.println("List of trades:");
		for(Trade n : trades){
			System.out.println(n);
		}
		System.out.println("List of candles:");
		List<Candlestick> candles = Candlestick.candlesticks(trades);
		for(Candlestick c : candles){
			System.out.println(c);
		}
	}
}
